xtask/src/release.rs

```diff
@@ -0,0 +1,162 @@
+    fn get_remote(&self) -> Result<String> {
+        let branch = cmd!("git rev-parse --abbrev-ref HEAD").read()?;
+        let remote = cmd!("git config branch.{branch}.remote").read()?;
+
+        if remote.is_empty() {
+            return Err("Could not get current git remote".into());
+        }
+
+        Ok(remote)
+    }
+
+    /// Get the current version of the crate from the manifest.
+    fn get_version(&self, project_root: &Path) -> Result<String> {
+        let manifest_toml = read_file(project_root.join(&self.name).join("Cargo.toml"))?;
+        let manifest: CargoManifest = from_toml_str(&manifest_toml)?;
+
+        Ok(manifest.package.version)
+    }
+}
+
+/// The required cargo manifest data of a crate.
+#[derive(Debug, Deserialize)]
+struct CargoManifest {
+    /// The package information.
+    package: CargoPackage,
+}
+
+/// The required package information from a crate's cargo manifest.
+#[derive(Debug, Deserialize)]
+struct CargoPackage {
+    /// The package version.
+    version: String,
+}
+
+/// String manipulations for crate release.
+trait StrExt {
+    /// Remove soft line breaks as defined in CommonMark spec.
+    fn trim_softbreaks(&self) -> String;
+}
+
+impl StrExt for str {
+    fn trim_softbreaks(&self) -> String {
+        let mut string = "".to_owned();
+        let mut current = 0;
+        let mut chars = self.char_indices();
+
+        while let Some(p) = self[current..].find('\n') {
+            let pos = current + p;
+            string.push_str(&self[current..pos]);
+
+            if self[pos..].starts_with("\n\n") {
+                // Keep new paragraphs (multiple `\n`s).
+                let next = self[pos..].find(|c: char| c != '\n').unwrap_or(0);
+                string.push_str(&self[pos..(pos + next)]);
+                current = pos + next;
+            } else if self[current..pos].ends_with("  ") || self[current..pos].ends_with('\\') {
+                // Keep hard line breaks (two spaces or a backslash before the line break)
+                string.push('\n');
+                current = pos + 1;
+            } else if let Some(p) = self[pos..].find(|c: char| !c.is_ascii_whitespace()) {
+                // Keep line break before list items (`\n` + whitespaces + `*` + whitespaces).
+                // Remove line break and keep one space otherwise.
+                let (_, char) = chars.find(|(i, _)| *i == pos + p).unwrap();
+                if char == '*' {
+                    match chars.next() {
+                        Some((_, next_char)) if next_char.is_ascii_whitespace() => {
+                            string.push('\n');
+                            current = pos + 1;
+                            continue;
+                        }
+                        _ => {}
+                    }
+                }
+                string.push(' ');
+                current = pos + p;
+            }
+        }
+
+        string + &self[(current..self.len())]
```

Comment:
Redundant parentheses, and I think `self.len()` is also not needed right?

```suggestion
        string + &self[current..]
```